Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp

```diff
@@ -107,6 +111,72 @@ namespace AZ
                 m_viewportContext.reset();
             }
 
+            //! Helper function that parses the command line arguments
+            //! looking for r_width, r_height and r_fullscreen.
+            //! It is important to call this before using r_width, r_height or r_fullscreen
+            //! because at the moment this system component initializes before Legacy System.cpp gets to parse
+            //! command line arguments into cvars.
+            static void UpdateCVarsFromCommandLine()
+            {
+                AZ::CommandLine* pCmdLine = nullptr;
+                ComponentApplicationBus::BroadcastResult(pCmdLine, &AZ::ComponentApplicationBus::Events::GetAzCommandLine);
+                if (!pCmdLine)
+                {
+                    return;
+                }
+
+                const AZStd::string fullscreenCvarName("r_fullscreen");
+                if (pCmdLine->HasSwitch(fullscreenCvarName))
+                {
+                    auto numValues = pCmdLine->GetNumSwitchValues(fullscreenCvarName);
+                    if (numValues > 0)
+                    {
+                        auto valueStr = pCmdLine->GetSwitchValue(fullscreenCvarName);
+                        if (AZ::StringFunc::LooksLikeBool(valueStr.c_str()))
+                        {
+                            r_fullscreen = AZ::StringFunc::ToBool(valueStr.c_str());
+                        }
+                    }
+                }
```

Comment:
Not a blocker but it would be really nice if these command line had a functions like this to avoid all this boiler plate code to query the value.
```
     bool GetBoolValue(AZStd::string_view switchName, bool default value = false);
     uint32_t GetUInt32Value(AZStd::string_view switchName, uint32_t default value = 1);
```

Reply:
Was thinking about that too. Will leave it for later.